pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -760,8 +761,25 @@ func (pl *dynamicResources) PostFilter(ctx context.Context, cs *framework.CycleS
 	// pick one claim randomly because there is no better heuristic.
 	for index := range state.unavailableClaims {
 		claim := state.claims[index]
+		if claim.Status.DeallocationRequested {
+			// Don't ask again, just wait.
+			break
+		}
```

Comment:
Shouldn't this be in `if len(claim.Status.ReservedFor) == 0 || len(claim.Status.ReservedFor) == 1 && claim.Status.ReservedFor[0].UID == pod.UID {}` ?

Reply:
No: if any claim is currently in the process of being deallocated, then the intend is to wait for that work to finish before deciding what to do next. Perhaps deallocating that one claim is enough to pick a node. If not, then the first attempt after completion will pick another claim.

Reply:
Your question got me thinking: it would be better to check *all* claims for `DeallocationRequested`. If the first claim is not deallocating and the second is, then the current check in the loop will trigger deallocation of the first claim.

/hold

Reply:
We can forget about all of this: if there is *any* claim for which deallocation is pending, PreFilter will mark the pod as unschedulable:
https://github.com/kubernetes/kubernetes/blob/33c5bd631d990f907f89f66cdf8088d252ab251b/pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go#L592-L595

This extra check here is unnecessary. I removed it.

Reply:
/hold cancel

Reply:
Right, that makes sense.